addons/controllers/webhooks/webhooks.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package webhooks provides functions to manage webhook TLS certificates
+package webhooks
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/webhooks"
+)
+
+type WebhookTLS struct {
+	Ctx           context.Context
+	K8sConfig     *rest.Config
+	CertPath      string
+	KeyPath       string
+	Name          string
+	ServiceName   string
+	LabelSelector string
+	Logger        logr.Logger
+	TLSSecret     *corev1.Secret
+	Namespace     string
+	InvalidWithIn time.Duration
+}
+
+func (w *WebhookTLS) Manage() error {
+	err := webhooks.ValidateTLSSecret(w.TLSSecret, w.InvalidWithIn)
+	if err != nil {
+		w.Logger.Info("Invalid webhook tls secret: " + err.Error())
+		w.Logger.Info("Installing new certificates")
+		w.TLSSecret, err = webhooks.InstallNewCertificates(w.Ctx, w.K8sConfig, w.CertPath, w.KeyPath, w.Name, w.Namespace, w.ServiceName, w.LabelSelector)
+		if err != nil {
+			return err
```

Comment:
question: do we want to put some logging before returning error ? Or it has ben handled by `InstallNewCertificates` already ?

Reply:
yeap. will add. InstallNewCertificates will not log, it returns error with info, so we should probably log it as you point out.

Reply:
Managed() is now meant to be called.  "err" will contain the error itself. 
It is up to the user (caller) to decide if the error should be logged or not. Otherwise we end up with multiple logs of the same error.  and since the what the error will be is not predictable we can just return it and let the calle handle the logging, as is done in ManageTLSCertificates